### PR TITLE
Upgrade terraform provider google beta to v4.70.0

### LINKED ISSUE #1097: Upgrade terraform-provider-google-beta to v4.69.0
Release details: https://github.com/hashicorp/terraform-provider-google-beta/releases/tag/v4.69.0

### LINKED ISSUE #1098: Upgrade terraform-provider-google-beta to v4.69.1
Release details: https://github.com/hashicorp/terraform-provider-google-beta/releases/tag/v4.69.1

### LINKED ISSUE #1100: Upgrade terraform-provider-google-beta to v4.70.0
Release details: https://github.com/hashicorp/terraform-provider-google-beta/releases/tag/v4.70.0

### LINKED ISSUE #1104: Upgrade terraform-provider-google-beta to v4.70.0
Test issue to trigger an upgrade

### PR DESCRIPTION
**Pull request by AaronFriel**
Fixes #1097 
Fixes #1098 
Fixes #1100 
Fixes #1104

The patches to the upstream were expanded into a set of multiple patches from the fork, one per commit, and then rebased using `FROM=v4.68.0 TO=v4.70.0 make upstream.rebase`.

Notably, two upstream data sources were patched out:
1. `google_organizations_iam_policy`
2. `google_organizations_iam_policy_legacy`

Automatic token mapping revealed that there were **two** resources that mapped to the Pulumi token `google:organizations/getIAMPolicy:getIAMPolicy`, and one resource which mapped to the same, but "Iam" instead of "IAM".

### CODE REVIEW

**Review by aq17 — APPROVED** [review]

File: patches/0009-Remove-duplicative-resource-token.patch [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+diff --git a/google-beta/provider.go b/google-beta/provider.go
+index cf8f23f49..52b5e5ddf 100644
+--- a/google-beta/provider.go
++++ b/google-beta/provider.go
+@@ -792,6 +792,12 @@ func Provider() *schema.Provider {
+ 
+ func DatasourceMap() map[string]*schema.Resource {
+ 	datasourceMap, _ := DatasourceMapWithErrors()
++
++	datasourcesToRemove := []string{"google_organization_iam_policy"}
```
Comment (thomas11) [inline patches/0009-Remove-duplicative-resource-token.patch:19]: The PR description mentions that `google_organizations_iam_policy_legacy` was also patched out but I don't see it here, is that somewhere else?

File: scripts/upstream.sh [hunk 2]
```diff
@@ -91,14 +91,12 @@ start_rebase() {
...
   for patch in ../patches/*.patch; do
     echo "Applying $patch"
-    if ! git apply --3way "$patch"; then
+    if ! git am --3way "$patch"; then
       echo
       echo "Failed to apply patch. Please run 'make upstream.rebase FROM=$TAG' where '$TAG' allows the patch set to apply cleanly"
       echo
       exit 1
     fi
-    patch=${patch#../patches/*-}
```
Comment (thomas11) [inline scripts/upstream.sh:100]: What's the new equivalent of this?

File: patches/0009-Remove-duplicative-resource-token.patch (hunk 1 above)
Comment (AaronFriel) [reply patches/0009-Remove-duplicative-resource-token.patch:19]: Will add that.

File: scripts/upstream.sh (hunk 2 above)
Comment (AaronFriel) [reply scripts/upstream.sh:100]: This extracted the filename of the patch to use as the commit message in the line below. `git am` preserves the existing commit message (subject & body).

See: https://github.com/pulumi/ci-mgmt/pull/469

**Pull request merged**